src/manager/monitor_proxy.c

```diff
@@ -0,0 +1,262 @@
+}
+
+ProxyMonitor *proxy_monitor_ref(ProxyMonitor *monitor) {
+        monitor->ref_count++;
+        return monitor;
+}
+
+void proxy_monitor_unref(ProxyMonitor *monitor) {
+        monitor->ref_count--;
+        if (monitor->ref_count != 0) {
+                return;
+        }
+
+        free(monitor->proxy_object_path);
+        free(monitor);
+}
+
+void proxy_monitor_close(ProxyMonitor *monitor) {
+        Manager *manager = monitor->manager;
+
+        if (monitor->subscription != NULL) {
+                manager_remove_subscription(manager, monitor->subscription);
+                subscription_unref(monitor->subscription);
+        }
+}
+
+int proxy_monitor_on_unit_new(
+                UNUSED void *userdata,
+                UNUSED const char *node,
+                UNUSED const char *unit,
+                UNUSED const char *reason) {
+        hirte_log_debug("ProxyMonitor processing unit new");
+        return 0;
+}
+
+int proxy_monitor_on_unit_state_changed(
+                UNUSED void *userdata,
+                UNUSED const char *node,
+                UNUSED const char *unit,
+                UNUSED const char *active_state,
+                UNUSED const char *substate,
+                UNUSED const char *reason) {
+        hirte_log_debug("ProxyMonitor processing unit state changed");
+        ProxyMonitor *monitor = (ProxyMonitor *) userdata;
+
+        int r = 0;
+        UnitActiveState state = active_state_from_string(active_state);
+        if (state == UNIT_ACTIVE || state == UNIT_FAILED) {
+                r = proxy_monitor_send_state(monitor, state);
+                proxy_monitor_close(monitor);
+                manager_remove_proxy_monitor(monitor->manager, monitor);
+        }
+        return r;
+}
+
+int proxy_monitor_on_unit_removed(
+                UNUSED void *userdata,
+                UNUSED const char *node,
+                UNUSED const char *unit,
+                UNUSED const char *reason) {
+        hirte_log_debug("ProxyMonitor processing unit removed");
+        return 0;
+}
+
+int proxy_monitor_on_unit_property_changed(
+                UNUSED void *userdata,
+                UNUSED const char *node,
+                UNUSED const char *unit,
+                UNUSED const char *interface,
+                UNUSED sd_bus_message *m) {
+        hirte_log_debug("ProxyMonitor processing unit property changed");
+        return 0;
+}
+
+int proxy_monitor_send_state(ProxyMonitor *monitor, UnitActiveState state) {
+        const char *unit_state = active_state_to_string(state);
+        hirte_log_debugf(
+                        "Updating node %s object %s with status %s",
+                        monitor->requesting_node->name,
+                        monitor->proxy_object_path,
+                        unit_state);
+
+        _cleanup_sd_bus_error_ sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_sd_bus_message_ sd_bus_message *message = NULL;
+        int r = sd_bus_call_method(
```

Comment:
This should be sd_bus_call_method_async. We don't want to block the main thread of the manager waiting for the response.